src/overlays/actors/ovl_En_An/z_en_an.c

```diff
@@ -13,160 +20,3438 @@
+            }
+            FALLTHROUGH;
+        case 0x2:
+        case 0x4:
+            Camera_SetTargetActor(Play_GetCamera(play, CutsceneManager_GetCurrentSubCamId(csId)), &this->actor);
+            ret = true;
+            this->msgEventState++;
+            break;
+
+        case 0x1:
+        case 0x3:
+            if (!CHECK_WEEKEVENTREG(WEEKEVENTREG_RECEIVED_ROOM_KEY) && (this->msgEventState == 3)) {
+                CutsceneManager_Stop(csId);
+                this->msgEventState = 5;
+            } else if ((this->actor.child != NULL) && (this->actor.child->update != NULL)) {
+                Camera_SetTargetActor(Play_GetCamera(play, CutsceneManager_GetCurrentSubCamId(csId)),
+                                      this->actor.child);
+            }
+            this->msgEventState++;
+            ret = true;
+            break;
+
+        case 0x5:
+            CutsceneManager_Stop(csId);
+            ret = true;
+            this->msgEventState++;
+            break;
+
+        default:
+            break;
+    }
+
+    return ret;
+}
+
+s32 EnAn_MsgEvent_GiveLunchToGranny(Actor* thisx, PlayState* play) {
+    EnAn* this = THIS;
+    s16 csId = EnAn_GetChildCsId(this, 0);
+    s32 ret = false;
+
+    switch (this->msgEventState) {
+        case 0x0:
+            ret = false;
+            if (!EnAn_ChangeCutscene(this, csId)) {
+                break;
+            }
+            FALLTHROUGH;
+        case 0x2:
+        case 0x4:
+        case 0x6:
+        case 0x8:
+            if ((this->actor.child != NULL) && (this->actor.child->update != NULL)) {
+                Camera_SetTargetActor(Play_GetCamera(play, CutsceneManager_GetCurrentSubCamId(csId)),
+                                      this->actor.child);
+            }
+            ret = true;
+            this->msgEventState++;
+            break;
+
+        case 0x1:
+        case 0x3:
+        case 0x5:
+        case 0x7:
+            Camera_SetTargetActor(Play_GetCamera(play, CutsceneManager_GetCurrentSubCamId(csId)), &this->actor);
+            ret = true;
+            this->msgEventState++;
+            break;
+
+        case 0x9:
+            CutsceneManager_Stop(csId);
+            ret = true;
+            this->msgEventState++;
+            break;
+
+        default:
+            break;
+    }
+
+    return ret;
+}
+
+// Only used if Player is using Kafei's Mask or if Human and Promised midnight meeting
+s32 EnAn_MsgEvent_MidnightMeeting(Actor* thisx, PlayState* play) {
+    EnAn* this = THIS;
+
+    if (this->msgEventState == 0) {
+        Player_SetCsActionWithHaltedActors(play, &this->actor, PLAYER_CSACTION_WAIT);
+        play->nextEntrance = ENTRANCE(STOCK_POT_INN, 3);
+        gSaveContext.nextCutsceneIndex = 0;
+        play->transitionTrigger = TRANS_TRIGGER_START;
+        play->transitionType = TRANS_TYPE_FADE_BLACK;
+        gSaveContext.nextTransitionType = TRANS_TYPE_FADE_BLACK_SLOW;
+        this->msgEventState++;
+    }
+
+    return 0;
```

Comment:
Should this be a boolean? The other `MsgEvent` functions seem to use booleans for `ret`.